File: python/1137.n-th-tribonacci-number.py
class Solution:
    def tribonacci(self, n: int) -> int:
        if n == 0:
            return 0

        if n == 1 or n == 2:
            return 1

        res = 0
        t_0 = 0
        t_1 = 1
        t_2 = 1

        for _ in range(2, n):
            res = t_0 + t_1 + t_2
            t_0 = t_1
            t_1 = t_2
            t_2 = res

        return res

File: golang/1137.n-th-tribonacci-number.go
func tribonacci(n int) int {
	if n == 0 {
		return 0
	}

	if n == 1 || n == 2 {
		return 1
	}

	res, t0, t1, t2 := 0, 0, 1, 1

	for i := 2; i < n; i++ {
		res = t0 + t1 + t2
		t0, t1, t2 = t1, t2, res
	}

	return res
}
